tests/test_slurm_command_gen_strategy.py

```diff
@@ -426,25 +426,53 @@ def setup_method(self):
 
     def assert_positional_lines(self, lines: list[str]):
         assert lines[0] == "#!/bin/bash"
-        assert lines[-6] == ""
-        assert lines[-5] == "export SLURM_JOB_MASTER_NODE=$(scontrol show hostname $SLURM_JOB_NODELIST | head -n 1)"
-        assert lines[-4] == ""
-        assert lines[-3] == self.env_vars_str
-        assert lines[-2] == ""
-        assert lines[-1] == self.srun_command
-
-    @pytest.mark.parametrize("missing_arg", ["job_name", "num_nodes", "partition", "node_list_str"])
+
+        assert f"#SBATCH --job-name={self.MANDATORY_ARGS['job_name']}" in lines
+        assert f"#SBATCH -N {self.MANDATORY_ARGS['num_nodes']}" in lines
+
+        partition = self.MANDATORY_ARGS.get("partition")
+        if partition:
+            assert f"#SBATCH --partition={partition}" in lines
+
+        node_list_str = self.MANDATORY_ARGS.get("node_list_str")
+        if node_list_str:
+            assert f"#SBATCH --nodelist={node_list_str}" in lines
+
+        gpus_per_node = self.MANDATORY_ARGS.get("gpus_per_node")
+        if gpus_per_node:
+            assert f"#SBATCH --gpus-per-node={gpus_per_node}" in lines
+            assert f"#SBATCH --gres=gpu:{gpus_per_node}" in lines
+
+        ntasks_per_node = self.MANDATORY_ARGS.get("ntasks_per_node")
+        if ntasks_per_node:
+            assert f"#SBATCH --ntasks-per-node={ntasks_per_node}" in lines
+
+        time_limit = self.MANDATORY_ARGS.get("time_limit")
+        if time_limit:
+            assert f"#SBATCH --time={time_limit}" in lines
+
+    @pytest.mark.parametrize("missing_arg", ["job_name", "num_nodes"])
     def test_raises_on_missing_args(self, missing_arg: str, strategy_fixture: SlurmCommandGenStrategy, tmp_path: Path):
         args = self.MANDATORY_ARGS.copy()
         del args[missing_arg]
 
         with pytest.raises(KeyError) as exc_info:
             strategy_fixture._write_sbatch_script(args, self.env_vars_str, self.srun_command, str(tmp_path))
-        assert f"KeyError('{missing_arg}')" in str(exc_info)
+        assert missing_arg in str(exc_info.value)
 
     def test_only_mandatory_args(self, strategy_fixture: SlurmCommandGenStrategy, tmp_path: Path):
+        # Ensure MANDATORY_ARGS includes all the arguments you expect to be present
+        args = {
+            "job_name": "test_job",
+            "num_nodes": 2,
+            "partition": "default",
+            "node_list_str": "node1,node2",
+        }
+
+        print(f"Args passed to the method: {args}")  # Debugging line
```

Comment:
Let's remove any print statements in pytest.

Reply:
Fixed